model/listener/ClientSyncLogListener.php

```diff
@@ -0,0 +1,145 @@
+ */
+
+namespace oat\taoSync\model\listener;
+
+use DateTime;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoSync\model\event\SyncFailedEvent;
+use oat\taoSync\model\event\SyncFinishedEvent;
+use oat\taoSync\model\event\SyncStartedEvent;
+use oat\taoSync\model\SyncLog\SyncLogDataHelper;
+use oat\taoSync\model\SyncLog\SyncLogDataParser;
+use oat\taoSync\model\SyncLog\SyncLogEntity;
+use oat\taoSync\model\SyncLog\SyncLogServiceInterface;
+
+/**
+ * Class ClientSyncLogListener
+ * @package oat\taoSync\model\listener
+ */
+class ClientSyncLogListener extends ConfigurableService
+{
+    const SERVICE_ID = 'taoSync/ClientSyncLogListener';
+
+    /**
+     * Create log record about started synchronization.
+     *
+     * @param SyncStartedEvent $event
+     */
+    public function logSyncStarted(SyncStartedEvent $event)
+    {
+        try {
+            $syncLogService = $this->getSyncLogService();
+            $params = $event->getSyncParameters();
+            $syncLogEntity = new SyncLogEntity(
+                $params[SyncLogServiceInterface::PARAM_SYNC_ID],
+                $params[SyncLogServiceInterface::PARAM_BOX_ID],
+                $params[SyncLogServiceInterface::PARAM_ORGANIZATION_ID],
+                $this->parseSyncData($event->getReport()),
+                SyncLogEntity::STATUS_IN_PROGRESS,
+                $event->getReport(),
+                new DateTime()
+            );
+
+            $syncLogService->create($syncLogEntity);
+        } catch (\Exception $e) {
+            $this->logError($e->getMessage());
+        }
+    }
+
+    /**
+     * Update log record for finished synchronization.
+     *
+     * @param SyncFinishedEvent $event
+     */
+    public function logSyncFinished(SyncFinishedEvent $event)
+    {
+        try {
+            $parameters = $event->getSyncParameters();
+            $syncLogEntity = $this->getSyncLogService()->getBySyncIdAndBoxId($parameters['sync_id'], $parameters['box_id']);
+
+            $report = $syncLogEntity->getReport();
+            $report->add($event->getReport());
+            if ($report->containsError()) {
+                $syncLogEntity->setFailed();
+            } else {
+                $syncLogEntity->setCompleted();
+            }
+            $eventData = $this->parseSyncData($event->getReport());
+            $newSyncData = SyncLogDataHelper::mergeSyncData($syncLogEntity->getData(), $eventData);
+
+            $syncLogEntity->setData($newSyncData);
+            $syncLogEntity->setFinishTime(new DateTime());
+
+            $this->getSyncLogService()->update($syncLogEntity);
+        } catch (\Exception $e) {
+            $this->logError($e->getMessage());
+        }
+    }
+
+    /**
+     * Update log record for failed synchronization.
+     *
+     * @param SyncFailedEvent $event
+     */
+    public function logSyncFailed(SyncFailedEvent $event)
+    {
+        try {
+            $parameters = $event->getSyncParameters();
```

Comment:
is it possible to make this block reusable?

Reply:
I'm not sure if it makes sense to move it to separate method